databricks/koalas/plot/matplotlib.py

```diff
@@ -536,35 +529,144 @@ def _get_bins(sdf, bins):
         return np.linspace(boundaries[0], boundaries[1], bins + 1)
 
     @staticmethod
-    def _compute_hist(sdf, bins):
+    def _compute_hist(kdf, bins):
         # 'data' is a Spark DataFrame that selects one column.
         assert isinstance(bins, (np.ndarray, np.generic))
 
-        colname = sdf.columns[-1]
+        sdf = kdf._internal.spark_frame
+        scols = []
+        for label in kdf._internal.column_labels:
+            scols.append(kdf._internal.spark_column_for(label))
+        sdf = sdf.select(*scols)
+
+        # 1. Make the bucket output flat to:
+        #     +----------+-------+
+        #     |__group_id|buckets|
+        #     +----------+-------+
+        #     |0         |0.0    |
+        #     |0         |0.0    |
+        #     |0         |1.0    |
+        #     |0         |2.0    |
+        #     |0         |3.0    |
+        #     |0         |3.0    |
+        #     |1         |0.0    |
+        #     |1         |1.0    |
+        #     |1         |1.0    |
+        #     |1         |2.0    |
+        #     |1         |1.0    |
+        #     |1         |0.0    |
+        #     +----------+-------+
+
+        def flat_bucket_df(left, right):
+            assert right is not None
+            if left is None:
+                return right.select(
+                    F.lit(group_id).alias("__group_id"), F.col(bucket_name).alias("__bucket")
+                )
+            else:
+                return left.union(
+                    right.select(
+                        F.lit(group_id).alias("__group_id"), F.col(bucket_name).alias("__bucket")
+                    )
+                )
 
-        bucket_name = "__{}_bucket".format(colname)
-        # creates a Bucketizer to get corresponding bin of each value
-        bucketizer = Bucketizer(
-            splits=bins, inputCol=colname, outputCol=bucket_name, handleInvalid="skip"
-        )
-        # after bucketing values, groups and counts them
+        colnames = sdf.columns
+        bucket_names = ["__{}_bucket".format(colname) for colname in colnames]
+
+        if LooseVersion(pyspark.__version__) >= LooseVersion("3.0.0"):
```

Comment:
I'm wondering whether we need this case? I guess we can just use the `else` case.

Reply:
Hm .. true. I don't think there's a notable benefit by doing this.